staging/src/k8s.io/apiserver/pkg/server/options/audit.go

```diff
@@ -20,28 +20,97 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"strings"
 
 	"github.com/spf13/pflag"
 	"gopkg.in/natefinch/lumberjack.v2"
 
+	"k8s.io/apiserver/pkg/audit"
 	"k8s.io/apiserver/pkg/audit/policy"
 	"k8s.io/apiserver/pkg/features"
 	"k8s.io/apiserver/pkg/server"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	pluginlog "k8s.io/apiserver/plugin/pkg/audit/log"
+	pluginwebhook "k8s.io/apiserver/plugin/pkg/audit/webhook"
 )
 
+func appendBackend(existing, newBackend audit.Backend) audit.Backend {
+	if existing == nil {
+		return newBackend
+	}
+	return audit.Union(existing, newBackend)
+}
+
+func advancedAuditingEnabled() bool {
+	return utilfeature.DefaultFeatureGate.Enabled(features.AdvancedAuditing)
+}
+
+type AuditOptions struct {
+	// Policy configuration file for filtering audit events that are captured.
+	// If unspecified, a default is provided.
+	PolicyFile string
+
+	// Plugin options
+
+	LogOptions     AuditLogOptions
+	WebhookOptions AuditWebhookOptions
+}
+
+// AuditLogOptions holds the legacy audit log writer. If the AdvancedAuditing feature
+// is enabled, these options determine the output of the structured audit log.
 type AuditLogOptions struct {
 	Path       string
 	MaxAge     int
 	MaxBackups int
 	MaxSize    int
+}
 
-	PolicyFile string
+// AuditWebhookOptions control the webhook configuration for audit events.
+type AuditWebhookOptions struct {
+	ConfigFile string
+	// Should the webhook asynchronous batch events to the webhook backend or
+	// should the webhook block responses?
+	//
+	// Defaults to asynchronous batch events.
+	Mode string
 }
 
-func NewAuditLogOptions() *AuditLogOptions {
-	return &AuditLogOptions{}
+func NewAuditOptions() *AuditOptions {
+	return &AuditOptions{
+		WebhookOptions: AuditWebhookOptions{Mode: pluginwebhook.ModeBatch},
+	}
+}
+
+func (o *AuditOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.PolicyFile, "audit-policy-file", o.PolicyFile,
+		"Path to the file that defines the audit policy configuration. Requires the 'AdvancedAuditing' feature gate."+
+			" With AdvancedAuditing, a profile is required to enable auditing.")
+
+	o.LogOptions.AddFlags(fs)
+	o.WebhookOptions.AddFlags(fs)
+}
+
+func (o *AuditOptions) ApplyTo(c *server.Config) error {
+	if o.PolicyFile == "" {
```

Comment:
Does this mean that legacy mode is disabled without a policy file? It shouldn't.